lib/express/download.js

```diff
@@ -0,0 +1,159 @@
+  return `${sid}${ENTITY_SEPARATOR}${timestamp}`;
+}
+
+/**
+ * @typedef {Object} EntityInfo
+ * @property {?string} sid - The hash value of the entry
+ * @property {?number} timestamp - The entry creation timestamp
+ */
+
+/**
+ * Parses the given entry name
+ *
+ * @param {string} name - The entry name string
+ * @return {EntityInfo} An empty object is returned if no values can be parsed
+ */
+function parseEntityName (name) {
+  if (!name.includes(ENTITY_SEPARATOR)) {
+    return {};
+  }
+  const [sid, timestampStr] = name.split(ENTITY_SEPARATOR);
+  const timestamp = parseInt(timestampStr, 10);
+  if (isNaN(timestamp)) {
+    return {sid};
+  }
+  return {sid, timestamp};
+}
+
+/**
+ * Stores a file from the local file system into the temporary
+ * downloads folder. The file is connected to a unique hash (Secure IDentifier),
+ * which is returned as the function result and can be used later
+ * to download the file from the server. The file is accessible for
+ * download within next 24 hours.
+ *
+ * @param {string} fullPath - Full path to the file
+ * @returns {string} The generated hash string
+ */
+async function storeFile (fullPath) {
+  if (!await fs.exists(fullPath)) {
+    throw new Error(`The file at '${fullPath}' does not exist or is not accessible`);
+  }
+  if (!await fs.exists(DOWNLOAD_DIR)) {
+    await mkdirp(DOWNLOAD_DIR);
+  }
+
+  const fileName = path.basename(fullPath);
+  const sid = crypto.createHash('sha256').digest('hex');
+  const timestamp = new Date().getTime();
+  const dirName = toEntityName(sid, timestamp);
+  await mkdirp(path.resolve(DOWNLOAD_DIR, dirName));
+  const storedFilePath = path.resolve(DOWNLOAD_DIR, dirName, fileName);
+  try {
+    _fs.createReadStream(fullPath).pipe(_fs.createWriteStream(storedFilePath));
+  } catch (err) {
+    throw new Error(`Cannot copy the file to downloads root. ` +
+                    `Is '${DOWNLOAD_DIR}' writeable?`);
+  }
+  return sid;
+}
+
+/**
+ * Matches the path to stored file with the given SID
+ *
+ * @param {string} sid The hash value
+ * @returns {?string} The relative path to the found file inside
+ * `DOWNLOAD_DIR` or `null` if no match has been found.
+ */
+async function sidToRelativeFilePath (sid) {
+  if (_.isEmpty(sid)) {
+    return null;
+  }
+  const matchedDirs = await fs.glob(path.resolve(DOWNLOAD_DIR, toEntityName(sid, '*')));
+  if (_.isEmpty(matchedDirs)) {
+    log.info(`There are no entities found for SID '${sid}'`);
+    return null;
+  }
+  const matchedFiles = await fs.readdir(_.head(matchedDirs));
+  if (_.isEmpty(matchedFiles)) {
+    log.info(`There are no matching files found for SID entity '${sid}'`);
+    return null;
+  }
+  return `${path.basename(_.head(matchedDirs))}/${_.head(matchedFiles)}`;
+}
+
+/**
+ * File download handler for the express server
+ *
+ * @param {http.ClientRequest} req The client request
+ * @param {http.ServerResponse} res The server response
+ */
+async function downloadFile (req, res) {
```

Comment:
for now it is not implemented as a "standard" endpoint. Would you prefer it to be available under `wd/hub`? In such case it will be necessary to add customised route handler, since this url behaves differently from other "normal" routes and returns response, which does not contain any standard headers.